clickhouse-http-client/src/main/java/com/clickhouse/client/http/ApacheHttpConnectionImpl.java

```diff
@@ -91,14 +96,31 @@ private CloseableHttpClient newConnection(ClickHouseConfig c) throws IOException
             r.register("https", socketFactory.create(c, SSLConnectionSocketFactory.class));
         }
 
-        HttpConnectionManager connManager = new HttpConnectionManager(r.build(), c);
+        long connectionTTL = config.getLongOption(ClickHouseClientOption.CONNECTION_TTL);
+        log.info("Connection TTL: %d ms", connectionTTL);
+        String poolReuseStrategy = c.getStrOption(ClickHouseHttpOption.CONNECTION_REUSE_STRATEGY);
+        PoolReusePolicy poolReusePolicy = PoolReusePolicy.LIFO;
+        if (poolReuseStrategy != null && !poolReuseStrategy.isEmpty()) {
+            try {
+                poolReusePolicy = PoolReusePolicy.valueOf(poolReuseStrategy);
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException("Invalid connection reuse strategy: " + poolReuseStrategy);
+            }
+        }
+        log.info("Connection reuse strategy: %s", poolReusePolicy.name());
+        HttpConnectionManager connManager = new HttpConnectionManager(r.build(), c, PoolConcurrencyPolicy.LAX,
+                poolReusePolicy, TimeValue.ofMilliseconds(connectionTTL));
         int maxConnection = config.getIntOption(ClickHouseHttpOption.MAX_OPEN_CONNECTIONS);
 
-        connManager.setMaxTotal(maxConnection);
+        connManager.setMaxTotal(Integer.MAX_VALUE); // unlimited on global level
```

Comment:
Can you explain why setting into MAX_VALUE

Reply:
if we have different routes then it will override max connection per route
previous version would work with only one route well